src/backends/sqlite3/statement.cpp

```diff
@@ -182,6 +182,9 @@ sqlite3_statement_backend::load_rowset(int totalRows)
                             col.buffer_.data_ = (col.buffer_.size_ > 0 ? new char[col.buffer_.size_] : NULL);
                             memcpy(col.buffer_.data_, sqlite3_column_blob(stmt_, c), col.buffer_.size_);
                             break;
+
+                        case dt_xml:
+                            throw soci_error("XML data type is not supported");
```

Comment:
Wouldn't it make more sense to gracefully handle it as strings? Otherwise it seems it would be difficult to write code using `dt_xml` and working with several databases.